### ISSUE TITLE
Metronome with changing meter

### ISSUE THREAD

**Issue by presidento** [paulrosen/abcjs#849]
When the meter changes, the metronome breaks, see [this example](https://editor.drawthedots.com/?t=%25%25MIDI%20drumon%0AQ%3A%20120%0AL%3A%201%2F4%0AM%3A%202%2F4%0A%25%25MIDI%20drum%20dd%2076%2077%2090%2090%0A%22_OK%22AD%20%7CAD%20%7C%0AM%3A%204%2F4%0A%25%25MIDI%20drum%20dddd%2076%2077%2077%2077%2090%2090%2090%2090%0A%22_Wrong%22A%2F2D%2F2D%2F2D%2F2%20z2%20%7C%5C%0A%25%25MIDI%20drum%20dddd%2076%2077%2077%2077%2090%2090%2090%2090%0A%22_OK%22ADDD%20%7C%20ADDD%20%7C%0AM%3A%202%2F4%0A%25%25MIDI%20drum%20dd%2076%2077%2090%2090%0A%22_Wrong%22zz%20%7C%5C%0AM%3A%202%2F4%0A%25%25MIDI%20drum%20dd%2076%2077%2090%2090%0A%22_OK%22AD%20%7C%20AD%20%7C%0AM%3A%203%2F4%0A%25%25MIDI%20drum%20ddd%2076%2077%2077%2090%2090%2090%0A%22_Wrong%22A2%2F3D2%2F3D2%2F3%20z%20%7C%5C%0A%25%25MIDI%20drum%20ddd%2076%2077%2077%2090%2090%2090%0A%22_OK%22ADD%20%7C%20ADD%20%7C):
```
%%MIDI drumon
Q: 120
L: 1/4
M: 2/4
%%MIDI drum dd 76 77 90 90
"_OK"AD |AD |
M: 4/4
%%MIDI drum dddd 76 77 77 77 90 90 90 90
"_Wrong"A/2D/2D/2D/2 z2 |\
%%MIDI drum dddd 76 77 77 77 90 90 90 90
"_OK"ADDD | ADDD |
M: 2/4
%%MIDI drum dd 76 77 90 90
"_Wrong"zz |\
M: 2/4
%%MIDI drum dd 76 77 90 90
"_OK"AD | AD |
M: 3/4
%%MIDI drum ddd 76 77 77 90 90 90
"_Wrong"A2/3D2/3D2/3 z |\
%%MIDI drum ddd 76 77 77 90 90 90
"_OK"ADD | ADD |
```

From the second bar it works perfectly, but I cannot make it working on the first bar when the meter changed.

The only workaround here is to switch off the metronome for the first bar, then switch back from the second, see [this example](https://editor.drawthedots.com/?t=%25%25MIDI%20drumon%0AQ%3A%20120%0AL%3A%201%2F4%0AM%3A%202%2F4%0A%25%25MIDI%20drum%20dd%2076%2077%2090%2090%0AAD%20%7CAD%20%7C%0AM%3A%204%2F4%0A%25%25MIDI%20drumoff%0AZ%20%7C%5C%0A%25%25MIDI%20drumon%0A%25%25MIDI%20drum%20dddd%2076%2077%2077%2077%2090%2090%2090%2090%0AADDD%20%7C%20ADDD%20%7C%0AM%3A%202%2F4%0A%25%25MIDI%20drumoff%0AZ%20%7C%5C%0AM%3A%202%2F4%0A%25%25MIDI%20drumon%0A%25%25MIDI%20drum%20dd%2076%2077%2090%2090%0AAD%20%7C%20AD%20%7C%0AM%3A%203%2F4%0A%25%25MIDI%20drumoff%0AZ%20%7C%5C%0A%25%25MIDI%20drumon%0A%25%25MIDI%20drum%20ddd%2076%2077%2077%2090%2090%2090%0AADD%20%7C%20ADD%20%7C):
```
%%MIDI drumon
Q: 120
L: 1/4
M: 2/4
%%MIDI drum dd 76 77 90 90
AD |AD |
M: 4/4
%%MIDI drumoff
Z |\
%%MIDI drumon
%%MIDI drum dddd 76 77 77 77 90 90 90 90
ADDD | ADDD |
M: 2/4
%%MIDI drumoff
Z |\
M: 2/4
%%MIDI drumon
%%MIDI drum dd 76 77 90 90
AD | AD |
M: 3/4
%%MIDI drumoff
Z |\
%%MIDI drumon
%%MIDI drum ddd 76 77 77 90 90 90
ADD | ADD |
```

**Comment by paulrosen** [comment]
I'll look into the rest of this, but I notice something odd in the 3/4 area: `A2/3`. What note duration is that? `3/4` is a dotted eighth.

**Comment by paulrosen** [comment]
I changed the example to just be rests so it was easy to hear:
```
%%MIDI drumon
Q: 120
L: 1/4
M: 2/4
%%MIDI drum dd 76 77 90 90
"_OK"z z |z z |
M: 4/4
%%MIDI drum dddd 76 77 77 77 90 90 90 90
"_Wrong"z z z z |\
%%MIDI drum dddd 76 77 77 77 90 90 90 90
"_OK"z z z z | z z z z |
M: 2/4
%%MIDI drum dd 76 77 90 90
"_Wrong"zz |\
M: 2/4
%%MIDI drum dd 76 77 90 90
"_OK"zz | zz |
M: 3/4
%%MIDI drum ddd 76 77 77 90 90 90
"_Wrong"z z z |\
%%MIDI drum ddd 76 77 77 90 90 90
"_OK"zzz | zzz |
```
Yes, that is definitely a bug. Thanks for the report.

**Comment by presidento** [comment]
I'd like to help fixing it. Can you give a hint how would you start working on it?

**Comment by paulrosen** [comment]
I'll poke around a little give you more of a hint, but there's a good chance that the only files involved are `abd_midi_sequencer.js` or `abc_midi_flattener.js`.

**Comment by presidento** [comment]
If it swapped out from your cache, do not bother yourself with it. I also
can do it. Thank you for the hint.

**Comment by paulrosen** [comment]
My eventual goal is to break all of the large files into smaller files. I won't touch those for at least a month, I'm sure.

**Comment by paulrosen** [comment]
Fixed in 6.2.0
